src/pages/index.js

```diff
@@ -53,38 +71,87 @@ const profilePopup = new PopupWithForm(
   config.popupProfileSelector,
   (userData) => {
     userInfo.setUserInfo(userData);
-    profilePopup.close();
+    profilePopup.notifyUserSaving(true);
+    api.updateUserInfo(userData).finally(() => {
+      profilePopup.notifyUserSaving(false);
+      profilePopup.close();
```

Comment:
You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) 


**Please, correct it everywhere**

Reply:
done